e2e/test/scenarios/models/reproductions/31406-model-link-formatting.cy.spec.js

```diff
@@ -0,0 +1,71 @@
+import { modal, popover, restore } from "e2e/support/helpers";
+import { PEOPLE_ID } from "metabase-types/api/mocks/presets";
+
+describe.skip("issue 31406", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+
+    cy.createQuestion(
+      {
+        dataset: true,
+        query: {
+          "source-table": PEOPLE_ID,
+          dataset: true,
+          limit: 2,
+        },
+      },
+      { visitQuestion: true },
+    );
+  });
+
+  it("should persist link text formatting settings for link column (metabase#31406)", () => {
+    cy.findByTestId("TableInteractive-root").findByText("Email").click();
+
+    popover().within(() => {
+      cy.icon("gear").click();
+    });
+
+    popover().within(() => {
+      cy.findByPlaceholderText("Link to {{bird_id}}")
+        .type("Link to {{name}}", {
+          parseSpecialCharSequences: false,
+        })
+        .blur();
+    });
```

Comment:
```suggestion
    popover().within(() => {
      cy.icon("gear").click();

      cy.findByPlaceholderText("Link to {{bird_id}}")
        .type("Link to {{name}}", {
          parseSpecialCharSequences: false,
        })
        .blur();
    });
```

This is perfectly fine.

Reply:
Btw, after #31666 gets merged we can (and we should) use more a11y friendly selectors. So instead of:
`cy.findByPlaceholderText("Link to {{bird_id}}")` we can use `cy.findByLabelText("Link text")`, which also conveys more meaning to the reader.